src/cinder/gl/GlslProg.cpp

```diff
@@ -426,6 +566,219 @@ void GlslProg::link()
+		uniformBlock.mBlockBinding = blockBinding;
+		
+		std::vector<GLint> uniformIndices( numActiveUniforms );
+		glGetActiveUniformBlockiv( mHandle, i, GL_UNIFORM_BLOCK_ACTIVE_UNIFORM_INDICES, uniformIndices.data() );
+		std::vector<GLuint> unSignedIndices( numActiveUniforms );
+		int j = 0;
+		for( auto & index : uniformIndices ) {
+			char name[512];
+			GLsizei nameLength;
+			GLint count;
+			GLenum type;
+			
+			glGetActiveUniform( mHandle, (GLuint)index, 511, &nameLength, &count, &type, name );
+			
+			UniformSemantic uniformSemantic = UniformSemantic::UNIFORM_USER_DEFINED;
+			
+			auto foundSemantic = semanticNameMap.find( name );
+			if( foundSemantic != semanticNameMap.end() ) {
+				uniformSemantic = foundSemantic->second;
+			}
+			
+			Uniform uniform;
+			uniform.mName		= name;
+			uniform.mLoc		= -1;
+			uniform.mIndex      = index;
+			uniform.mCount		= count;
+			uniform.mType		= type;
+			uniform.mSemantic	= uniformSemantic;
+			unSignedIndices[j++]= index;
+			
+			uniformBlock.mActiveUniforms.push_back( uniform );
+		}
+
+		std::vector<GLint> uniformOffset( numActiveUniforms );
+		glGetActiveUniformsiv( mHandle,
+							  numActiveUniforms,
+							  unSignedIndices.data(),
+							  GL_UNIFORM_OFFSET,
+							  uniformOffset.data() );
+		std::vector<GLint> uniformArrayStride( numActiveUniforms );
+		glGetActiveUniformsiv( mHandle,
+							  numActiveUniforms,
+							  unSignedIndices.data(),
+							  GL_UNIFORM_ARRAY_STRIDE,
+							  uniformArrayStride.data() );
+		std::vector<GLint> uniformMatrixStride( numActiveUniforms );
+		glGetActiveUniformsiv( mHandle,
+							  numActiveUniforms,
+							  unSignedIndices.data(),
+							  GL_UNIFORM_MATRIX_STRIDE,
+							  uniformMatrixStride.data() );
+		
+		uniformBlock.mActiveUniformInfo.insert( make_pair( GL_UNIFORM_OFFSET, uniformOffset ) );
+		uniformBlock.mActiveUniformInfo.insert( make_pair( GL_UNIFORM_ARRAY_STRIDE, uniformArrayStride ) );
+		uniformBlock.mActiveUniformInfo.insert( make_pair( GL_UNIFORM_MATRIX_STRIDE, uniformMatrixStride ) );
+		
+		mUniformBlocks.push_back( uniformBlock );
+	}
+}
+
+void GlslProg::cacheActiveTransformFeedbackVaryings()
+{
+	GLint numActiveTransformFeedbackVaryings;
+	glGetProgramiv( mHandle, GL_TRANSFORM_FEEDBACK_VARYINGS, &numActiveTransformFeedbackVaryings );
+	
+	if( numActiveTransformFeedbackVaryings > 0 ) {
+		for( int i = 0; i < numActiveTransformFeedbackVaryings; i++ ) {
+			GLchar name[500];
+			GLint length;
+			GLsizei count;
+			GLenum type;
+			
+			glGetTransformFeedbackVarying( mHandle, i, 500, &length, &count, &type, name );
+			name[length] = 0;
+			
+			// So they've told us at setup what to look for. Find the varying, they told us
+			// about so that we can finish the setup.
+			auto varying = findTransformFeedbackVaryings( name );
+			if( varying ) {
+				varying->mCount = count;
+				varying->mType = type;
+			}
+		}
+		
+		// TODO: This needs to be created correctly and checked, this is where the magic will happen.
```

Comment:
It isn't going to be tackled with this PR. I could probably remove some of the psuedo code. However, this would be where Cinder could make connections between output (transform feedback) buffers and input (attribute) buffers. There's quite a bit of checking we can do, similar to the checks on Uniform or the AttributeMap check inside of buildVao, that could give feedback to the user if they've set things up incorrectly. I don't have a problem removing it though.